### PR TITLE
Add warning for unwrapped coordinates in MSD module

### LINKED ISSUE #3074: Improve docs for MSD class and add detailed example to userguide.
Clearer docs around what best practices are for calculation of MSDs are probably needed. For example a more prominent note that you should use unwrapped trajectories is needed.

There is also a reasonable amount of assumed knowledge in the docs as they are currently and it seems to be tripping people up. The language should be improved to be more accessible. A detailed example in the user guide would be handy in this respect.

### PR DESCRIPTION
**Pull request by hmacdope**
Fixes #3074

Changes made in this Pull Request:
 - Add a warning suggesting the use of unwrapped coordinates to the MSD docs.

PR Checklist
------------

 - [X] Docs?
 - [X] CHANGELOG updated?
 - [X] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/analysis/msd.py [hunk 1]
```diff
@@ -62,6 +62,14 @@
 Please cite [Calandri2011]_ [Buyl2018]_ if you use this module in addition to
 the normal MDAnalysis citations.
 
+.. warning::
+    To correctly compute the MSD using this analysis module, you must supply
+    coordinates in the **unwrapped** convention. That is, when they pass
```
Comment (IAlibay) [inline package/MDAnalysis/analysis/msd.py:67]: probably best to rephrase "they" to "atoms"?

**Review by IAlibay — CHANGES_REQUESTED** [review]
Thanks @hmacdope, couple of comments.

Personally, I'd skip the changelog completely, it's kinda self-documented (if that makes sense). We can keep it if you feel strongly about it though.

File: package/MDAnalysis/analysis/msd.py (hunk 1 above)
Comment (hmacdope) [reply package/MDAnalysis/analysis/msd.py:67]: 👍

File: package/MDAnalysis/analysis/msd.py [hunk 2]
```diff
@@ -62,6 +62,14 @@
...
+    the periodic boundary, they must not be **wrapped** back into the primary
+    simulation cell. Various simulation packages provide utilities to convert
```
[the remark this answers is not in the document]
Comment (hmacdope) [reply package/MDAnalysis/analysis/msd.py:69]: Our own unwrapping code does not have the correct transformation implemented as of now. I get that there is a bit of a naming clash here. What we need is the equivalent of the unwrapping scheme applied by `gmx trjconv -pbc nojump` or related transform.

[the remark this answers is not in the document]
Comment (IAlibay) [reply package/MDAnalysis/analysis/msd.py:69]: I see. Can you add a note here to that effect? Wouldn't want users reading this and assuming that they can just applying unwrap transformations to make this work.

[the remark this answers is not in the document]
Comment (hmacdope) [reply package/MDAnalysis/analysis/msd.py:69]: Great idea.

**Comment by hmacdope** [comment]
We happy with the warning message now?

**Review by IAlibay — APPROVED** [review]

**Pull request merged**